### PR TITLE
Fix CORS throwing 500 upon encountering a malformed URL #33682

### PR DESCRIPTION
**Pull request by IgorOffline**
Fix for [33682](https://github.com/spring-projects/spring-framework/issues/33682)

Also remove BeforeEach in CrossOriginTests because many tests do not need the "setup" step.

### CODE REVIEW

File: spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java [hunk 1]
```diff
@@ -38,19 +39,22 @@ public abstract class CorsUtils {
 	 * Returns {@code true} if the request is a valid CORS one by checking {@code Origin}
 	 * header presence and ensuring that origins are different.
 	 */
-	public static boolean isCorsRequest(HttpServletRequest request) {
+	public static IsCorsRequestResult isCorsRequest(HttpServletRequest request) {
```
Comment (sfc-gh-jzana) [inline spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java:42]: You need to update the javadocs.

Comment (sfc-gh-jzana) [inline spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java:42]: Is it OK to introduce a breaking public API change for this bugfix?  If so, it's probably worth documenting in the release notes.

File: spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java [hunk 2]
```diff
@@ -38,19 +39,22 @@ public abstract class CorsUtils {
...
 		String origin = request.getHeader(HttpHeaders.ORIGIN);
 		if (origin == null) {
-			return false;
+			return IsCorsRequestResult.IS_NOT_CORS_REQUEST;
+		}
+		if (!(new UrlValidator().isValid(origin))) {
```
Comment (sfc-gh-jzana) [inline spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java:47]: Is this the right validation to use, given https://github.com/spring-projects/spring-framework/issues/33639 ?

File: spring-web/src/main/java/org/springframework/web/cors/DefaultCorsProcessor.java [hunk 3]
```diff
@@ -83,9 +83,14 @@ public boolean processRequest(@Nullable CorsConfiguration config, HttpServletReq
 			response.addHeader(HttpHeaders.VARY, HttpHeaders.ACCESS_CONTROL_REQUEST_HEADERS);
 		}
 
-		if (!CorsUtils.isCorsRequest(request)) {
+		IsCorsRequestResult isCorsRequestResult = CorsUtils.isCorsRequest(request);
```
Comment (sfc-gh-jzana) [inline spring-web/src/main/java/org/springframework/web/cors/DefaultCorsProcessor.java:86]: Note: there is another caller of this function: `org.springframework.web.cors.reactive.DefaultCoreProcessor`.

File: spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java (hunk 2 above)
Comment (IgorOffline) [reply spring-web/src/main/java/org/springframework/web/cors/CorsUtils.java:47]: I wasn't aware of already existing validation mechanisms - will change ASAP

**Comment by IgorOffline** [comment]
Thank you!

**Pull request closed without merging**